### PR TITLE
acquisition: link order line to a document

### PR DESCRIPTION
**Pull request by lauren-d**
* Adds a link between order line and document.
* Removes description field in order line resource

Co-Authored-by: Lauren-D <laurent.dubois@itld-solutions.be>

## Why are you opening this PR?

- Which task/US does it implement?
https://tree.taiga.io/project/rero21-reroils/task/1276?kanban-status=1224894

## How to test?

- Should be tested with https://github.com/rero/rero-ils-ui/pull/124

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json [hunk 1]
```diff
@@ -150,6 +145,18 @@
         }
       }
     },
+    "document": {
```
Comment (jma) [inline rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json]: description is missing. This is the case for almost all the field in this schema. I would be great to have it.

File: rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json [hunk 2]
```diff
@@ -19,8 +19,8 @@
     "pid",
     "acq_account",
     "acq_order",
-    "amount",
-    "description"
+    "document",
```
Comment (jma) [inline rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json]: How the document is required? Is that means that I have to catalog the document before buying it?

File: rero_ils/modules/documents/api.py [hunk 3]
```diff
@@ -127,6 +128,12 @@ def get_number_of_loans(self):
         results = search.source().count()
         return results
 
+    def get_number_of_acquisition_order_lines(self):
+        """Get number of acquisition order lines for document."""
+        results = AcqOrderLinesSearch().filter(
```
Comment (jma) [inline rero_ils/modules/documents/api.py]: variable is not very useful here

File: rero_ils/modules/documents/api.py [hunk 4]
```diff
@@ -136,6 +143,9 @@ def get_links_to_me(self):
         loans = self.get_number_of_loans()
         if loans:
             links['loans'] = loans
+        acquisition_order_lines = self.get_number_of_acquisition_order_lines()
```
Comment (jma) [inline rero_ils/modules/documents/api.py]: Order lines or number of order lines?

File: rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json (hunk 2 above)
Comment (lauren-d) [reply rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json]: Yes, we need to catalog the document before order it.

File: rero_ils/modules/documents/api.py (hunk 3 above)
Comment (lauren-d) [reply rero_ils/modules/documents/api.py]: done

File: rero_ils/modules/documents/api.py (hunk 4 above)
Comment (lauren-d) [reply rero_ils/modules/documents/api.py]: modified to number_of_order_lines

File: rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json (hunk 1 above)
Comment (lauren-d) [reply rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json]: yes, the description field is missing in many resources schema.
Maybe we can ask the missing description to the PO group?

File: tests/api/test_acq_order_lines_rest.py [hunk 5]
```diff
@@ -157,6 +157,15 @@ def test_acq_order_lines_can_delete(client, acq_order_line_fiction_martigny):
     assert not reasons
 
 
+def test_acq_order_lines_document_can_delete(
+        client, document, acq_order_line_fiction_martigny):
+    """Test can delete a document with a linked acquisition order line."""
+    assert not document.can_delete
+
+    reasons = document.reasons_not_to_delete()
+    assert reasons
```
Comment (BadrAly) [inline tests/api/test_acq_order_lines_rest.py]: assert `acq_order_lines` in reasons?

File: rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json [hunk 6]
```diff
@@ -150,6 +143,18 @@
...
     },
+    "document": {
+      "title": "Document",
+      "type": "object",
+      "properties": {
+        "$ref": {
+          "title": "Document URI",
+          "type": "string",
+          "pattern": "^https://ils.rero.ch/api/documents/.+?$",
+          "default": "https://ils.rero.ch/api/documents/"
```
Comment (Garfield-fr) [inline rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json]: This default isn't valid

Comment (lauren-d) [reply rero_ils/modules/acq_order_lines/jsonschemas/acq_order_lines/acq_order_line-v0.0.1.json]: done

File: tests/api/test_acq_order_lines_rest.py (hunk 5 above)
Comment (lauren-d) [reply tests/api/test_acq_order_lines_rest.py]: done

**Pull request merged**